crates/api/src/event/event_adapter.rs

```diff
@@ -0,0 +1,107 @@
+use std::sync::{mpsc, Arc, Mutex};
+
+use dces::entity::Entity;
+
+use crate::{event::*, shell::WindowRequest};
+
+/// The `EventAdapter` provides a thread save way to push events to the widget tree of a window.
+///
```

Comment:
Do you mean thread-safe? e.g. sharing data safely between threads without causing a data race?

Reply:
yes